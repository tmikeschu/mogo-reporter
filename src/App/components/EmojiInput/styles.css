.emoji {
  font-size: 3rem;
}

.emoji input {
  display: none;
}

.emoji :hover {
  font-size: 3.2rem;
  transition: all 0.2s ease;
  cursor: pointer;
}

.emoji--selected-true {
  background-color: red;
  font-size: 3.2rem;
  transition: all 0.5s ease;
}
