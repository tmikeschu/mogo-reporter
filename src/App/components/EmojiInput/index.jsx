import React from "react";
import "./styles.css";

const EmojiInput = ({ input, parent, selected, onClick }) => {
  return (
    <span className={"emoji"}>
      <label
        className={`emoji--selected-${input.name === selected}`}
        htmlFor={`${input.name}`}
      >
        <span>{input.emoji}</span>
      </label>
      <input
        type="radio"
        data-metric={parent}
        id={`${input.name}`}
        onClick={onClick}
        value={input.name}
        checked={input.name === selected}
      />
    </span>
  );
};

export default EmojiInput;
