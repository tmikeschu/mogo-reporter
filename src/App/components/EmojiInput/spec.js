import React from "react";
import { mount } from "enzyme";
import EmojiInput from "./";

describe("<EmojiInput />", () => {
  it("mounts without crashing", () => {
    let input = { name: "hello", emoji: "hi" };
    let onClick = jest.fn();
    const wrapper = mount(
      <EmojiInput input={input} parent="bar" selected="foo" onClick={onClick} />
    );
    expect(wrapper).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
