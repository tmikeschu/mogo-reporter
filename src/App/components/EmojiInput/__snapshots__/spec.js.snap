// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmojiInput /> mounts without crashing 1`] = `
<EmojiInput
  input={
    Object {
      "emoji": "hi",
      "name": "hello",
    }
  }
  onClick={[Function]}
  parent="bar"
  selected="foo"
>
  <span
    className="emoji"
  >
    <label
      className="emoji--selected-false"
      htmlFor="hello"
    >
      <span>
        hi
      </span>
    </label>
    <input
      checked={false}
      data-metric="bar"
      id="hello"
      onClick={[Function]}
      type="radio"
      value="hello"
    />
  </span>
</EmojiInput>
`;
